compiler/noirc_frontend/src/tests.rs

```diff
@@ -0,0 +1,1158 @@
+            }
+            result.push(cur_capture);
+
+            // Check for other captures recursively within the lambda body
+            let hir_body_expr = interner.expression(&lambda_expr.body);
+            if let HirExpression::Block(block_expr) = hir_body_expr {
+                parse_statement_blocks(block_expr.statements(), interner, result);
+            }
+        }
+    }
+
+    #[test]
+    fn resolve_empty_function() {
+        let src = "
+            fn main() {
+
+            }
+        ";
+        assert!(get_program_errors(src).is_empty());
+    }
+    #[test]
+    fn resolve_basic_function() {
+        let src = r#"
+            fn main(x : Field) {
+                let y = x + x;
+                assert(y == x);
+            }
+        "#;
+        assert!(get_program_errors(src).is_empty());
+    }
+    #[test]
+    fn resolve_unused_var() {
+        let src = r#"
+            fn main(x : Field) {
+                let y = x + x;
+                assert(x == x);
+            }
+        "#;
+
+        let errors = get_program_errors(src);
+        assert!(errors.len() == 1, "Expected 1 error, got: {:?}", errors);
+        // It should be regarding the unused variable
+        match &errors[0].0 {
+            CompilationError::ResolveError(ResolverError::UnusedVariable { ident }) => {
+                assert_eq!(&ident.0.contents, "y");
+            }
+            _ => unreachable!("we should only have an unused var error"),
+        }
+    }
+
+    #[test]
+    fn resolve_unresolved_var() {
+        let src = r#"
+            fn main(x : Field) {
+                let y = x + x;
+                assert(y == z);
+            }
+        "#;
+        let errors = get_program_errors(src);
+        assert!(errors.len() == 1, "Expected 1 error, got: {:?}", errors);
+        // It should be regarding the unresolved var `z` (Maybe change to undeclared and special case)
+        match &errors[0].0 {
+            CompilationError::ResolveError(ResolverError::VariableNotDeclared {
+                name,
+                span: _,
+            }) => assert_eq!(name, "z"),
+            _ => unimplemented!("we should only have an unresolved variable"),
+        }
+    }
+
+    #[test]
+    fn unresolved_path() {
+        let src = "
+            fn main(x : Field) {
+                let _z = some::path::to::a::func(x);
+            }
+        ";
+        let errors = get_program_errors(src);
+        assert!(errors.len() == 1, "Expected 1 error, got: {:?}", errors);
+        for (compilation_error, _file_id) in errors {
+            match compilation_error {
+                CompilationError::ResolveError(err) => {
+                    match err {
+                        ResolverError::PathResolutionError(PathResolutionError::Unresolved(
+                            name,
+                        )) => {
+                            assert_eq!(name.to_string(), "some");
```

Comment:
Before refactoring the test expected `func`, but after I refactor it to use `get_program_errors` it expects `some`

This is change in test behavior and I think the new behavior is the correct one. Still, it got me thinking, so I just want to be sure here